### ISSUE TITLE
[qt6/readability-js] local html pages can't display external images

### ISSUE THREAD

**Issue by rien333** [qutebrowser/qutebrowser#7206]
**Version**

```
qutebrowser v2.5.0
Git commit: 64e8c7132 on makepkg (2022-05-26 14:37:27 +0200)
Backend: QtWebEngine 6.3, based on Chromium 94.0.4606.126
Qt: 6.3.0
```

**Description**

The readability-js userscript (and the other readability scripts) store their output in a html file in `~/.local/share/qutebrowser/userscripts`, which the browser then loads. Probably with the introduction of Qt6, any references to images in local html files throw a `Failed to load resource: net::ERR_NETWORK_ACCESS_DENIED` in the DevTools console. My hunch is that this all relates to [this commit in qtwebengine6](https://code.qt.io/cgit/qt/qtwebengine.git/commit/?id=3071de1e07be28d763164a037d946281146bf31d). In short, this blockage results in no images being shown on "file://..." pages.

People with locally stored homepages might also be affected.

What's interesting is that other browsers don't show this same problem, even with the exact same html. The html the firefox readability script outputs fails to display and load images when you open it in qutebrowser. Reversedly, the html the qutebrowser userscript produces loads and displays fine in firefox.

Pointers are very welcome, though I might eventually figure this out myself. I was thinking we could maybe send some information in the html header that tells webengine that there's nothing to worry about? (the commit above mentions something about `Access-Control-Allow-Origin`)

**How to reproduce**

Save the following html to a file, and open it in qutebrowser:

``` html
<html>
<p><img src="https://upload.wikimedia.org/wikipedia/commons/thumb/c/cf/Finding_the_median.png/220px-Finding_the_median.png"></p>
<p>Test</p>
</html>
```

Note that this very same file will load fine in chromium and firefox (e.g. by opening `file:///tmp/test.html`).

**Comment by toofar** [comment]
Are you on the latest version of that branch? @The-Compiler just pushed a workaround for https://github.com/qutebrowser/qutebrowser/issues/7194 a few days ago, possibly this is the same issue.

**Comment by rien333** [comment]
Whoops, sorry I missed that obviously related issue. But yes, unfortunately I'm on the latest commit of the qt6 branch (https://github.com/qutebrowser/qutebrowser/commit/64e8c713239275fb24f949b013e51cfb4d9c77a3).

Still, setting `content.local_content_can_access_remote_urls` to `true` fixes the problem. If that is the best workaround/solution available, I suppose this can be closed, and I will just add a note to the readability userscript(s).

**Comment by toofar** [comment]
I see. We probably want to keep this open then until we know the reason for the change. It might be an(other) unintended regression, especially since it works fine on other browsers.

**Comment by The-Compiler** [comment]
It seems odd that QtWebEngine now defaults to a much stricter set of rules than what common browsers do, and also doesn't have a way to disable those stricter rules without letting local pages access *all* remote resources...

Perhaps we should just set that setting for `true` for the qutebrowser data dir by default?

**Comment by rien333** [comment]
> It seems odd that QtWebEngine now defaults to a much stricter set of rules than what common browsers do

Yeah. 

Arch just pushed [this patch](https://code.qt.io/cgit/qt/qtwebengine.git/patch/?id=8d0bd4b1), which seemingly relates to this overly restrictive ruleset. I didn't take the time to comprehend the patch, but, unfortunately, it doesn't seem to fix this particular issue. 

> Perhaps we should just set that setting for true for the qutebrowser data dir by default?

Seems fine to me. It's not like any of the scripts that store (html) data in that folder also run external javascript.

**Comment by The-Compiler** [comment]
The Archlinux backport is for clicking links on local pages, see #7194.

**Comment by The-Compiler** [comment]
Should be fixed via 8d35fcdc187ca1578466206f98bec80b1ed44553 by overriding that setting by default for `~/.local/share/qutebrowser/userscripts/*` (or wherever the data dir is).

**Comment by rien333** [comment]
Thank you!
